### PR TITLE
Codegen: immutable value objects with auto-generated with*: and keyword constructor (BT-923)

### PR DESCRIPTION
**Pull request by jamesc**
## Summary

- Auto-generate getter, `with*:` functional setter, and all-fields keyword constructor for each slot in `Value subclass:` classes
- Register keyword constructor in the class actor's `classMethods` map so `class_send` dispatch routes correctly
- Skip auto-generation for slots where the user has explicitly defined a matching method

## Key Changes

- `value_type_codegen.rs`: new `AutoSlotMethods` struct + `compute_auto_slot_methods`, `generate_slot_getter`, `generate_slot_setter`, `generate_keyword_constructor_fn`, `generate_auto_slot_dispatch_arms` helpers; updated `generate_primitive_dispatch` and `generate_primitive_has_method` to route auto-generated selectors
- `gen_server/methods.rs`: `generate_register_class` now includes keyword constructor in `classMethods` for `ClassKind::Value` classes
- 12 new unit tests covering exports, function bodies, dispatch routing, `has_method` inclusion, and user-override behaviour
- New stdlib BUnit test `value_subclass_test.bt` and fixture `value_point.bt` for end-to-end coverage

https://linear.app/beamtalk/issue/BT-923

🤖 Generated with [Claude Code](https://claude.com/claude-code)

## Summary by CodeRabbit

* **New Features**
  * Auto-generated getters and functional setters for Value subclass state fields.
  * Auto-generated keyword constructors for Value subclasses with non-empty state, skipping generation on selector conflicts or when user overrides exist; Object/Actor subclasses unaffected.
* **Tests**
  * Added extensive tests and fixtures validating auto-generated accessors, keyword constructors, dispatch/routing, override suppression, and edge cases.

### CODE REVIEW

File: crates/beamtalk-core/src/codegen/core_erlang/gen_server/methods.rs [hunk 1]
```diff
@@ -1107,6 +1107,36 @@ impl CoreErlangGenerator {
...
+                    let kw_sel = expected_kw_sel;
+                    let kw_arity = class.state.len();
+                    class_method_entries.push(docvec![
+                        "'",
+                        Document::String(kw_sel),
+                        "' => ~{'arity' => ",
+                        Document::String(kw_arity.to_string()),
+                        "}~",
+                    ]);
+                }
```
[the remark this answers is not in the document]
Comment (jamesc) [reply crates/beamtalk-core/src/codegen/core_erlang/gen_server/methods.rs]: Fixed in commit 58b57736. Added a `collides_with_builtin` check that guards against the keyword-constructor selector matching any built-in class selector (`new` or `new:`). The condition now reads: `if !collides_with_builtin && !kw_selector_already_defined`.

File: crates/beamtalk-core/src/codegen/core_erlang/value_type_codegen.rs [hunk 2]
```diff
@@ -412,6 +571,111 @@ impl CoreErlangGenerator {
...
+        }
+
+        docvec![
+            Document::String(format!(
+                "'class_{kw_selector}'/{arity} = fun ({params}) ->\n"
+            )),
+            Document::String(format!("    ~{{{map_fields}}}~\n")),
+            "\n",
+        ]
+    }
```
[the remark this answers is not in the document]
Comment (jamesc) [reply crates/beamtalk-core/src/codegen/core_erlang/value_type_codegen.rs]: Fixed in commit 58b57736. Rewrote `generate_slot_getter`, `generate_slot_setter`, `generate_keyword_constructor_fn`, and `generate_auto_slot_dispatch_arms` to use `docvec\!`/`Document` composition with `concat()` for variable-length map fields. The `use std::fmt::Write as _` import is also removed since `write\!()` is no longer used.
